FlowCrypt/Controllers/Inbox/InboxViewController.swift

```diff
@@ -174,14 +191,28 @@ extension InboxViewController {
         }
     }
 
-    private func fetchAndRenderEmailsOnly(_ batchContext: ASBatchContext?) {
+    internal func getSearchQuery() -> String? {
+        var searchQuery: String?
+        if searchedExpression.isNotEmpty {
+            searchQuery = "\(searchedExpression) OR subject:\(searchedExpression)"
+        }
+        return searchQuery
+    }
+
+    internal func fetchAndRenderEmailsOnly(_ batchContext: ASBatchContext?) {
         Task {
             do {
-                state = .fetching
+                if isSearch {
+                    state = .searching
+                    await self.tableNode.reloadData()
+                } else {
+                    state = .fetching
+                }
                 let context = try await service.fetchInboxItems(
                     using: FetchMessageContext(
-                        folderPath: viewModel.path,
+                        folderPath: isSearch ? nil : viewModel.path, // pass nil in search screen to search for all folders
```

Comment:
@tomholub In search screen, Do we have to search for all folders or just current folder?
I thought it should search for all folders so I added this logic.
Gmail and previous search screen also searched for all folders though.

Reply:
Aligning functionality with Gmail is usually the right thing.

Reply:
ok